c10/core/Scalar.h

```diff
@@ -316,7 +361,19 @@ class C10_API Scalar {
   // We can't set v in the initializer list using the
   // syntax v{ .member = ... } because it doesn't work on MSVC
  private:
-  enum class Tag { HAS_d, HAS_i, HAS_z, HAS_b, HAS_sd, HAS_si, HAS_sb };
+  enum class Tag { HAS_d, HAS_i, HAS_u, HAS_z, HAS_b, HAS_sd, HAS_si, HAS_sb };
+
+  // Note [Meaning of HAS_u]
```

Comment:
From reading this again, another option could be calling it `HAS_li` for large integer to avoid introducing the idea of unsigned if we want to only have the idea that this is for very larger integer.
Happy with the current version as well, so feel free to do either.

Reply:
I guess it wouldn't even be that hard now that we have symint. It would be the same logic as the on used for large negative ints :D

Reply:
Yeah, exactly. Use the SymInt field to point to an arbitrary size integer representation. Honestly, hardest part is picking the bignum library you want.